tools/data_source/upload.xml

```diff
@@ -21,16 +21,17 @@
         #end if
         '${output.dataset.dataset.id}:${output.files_path}:${file_name}'
     #end while
-  </command>
+  ]]></command>
   <inputs nginx_upload="true">
-    <param name="file_type" type="select" label="File Format" help="Which format? See help below">
+    <!-- <param name="file_type" type="select" label="File Format" help="Which format? See help below">
```

Comment:
I'm surprised Planemo even uses upload.xml and not data fetch at this point. I'm fine removing this block though since it isn't used by the UI anymore and the replaced block is more general.

Reply:
If you check any planemo run, e.g. at IUC then you find lines like 

```
2022-01-14 15:57:29,127 INFO  [galaxy.jobs.command_factory] Built script [/tmp/tmpn5toshnh/job_working_directory/000/1/tool_script.sh] for tool command [python '/tmp/tmpn5toshnh/galaxy-dev/tools/data_source/upload.py' '/tmp/tmpn5toshnh/galaxy-dev' '/tmp/tmpn5toshnh/job_working_directory/000/1/registry.xml' '/tmp/tmpn5toshnh/job_working_directory/000/1/upload_params.json' '1:/tmp/tmpn5toshnh/job_working_directory/000/1/working/dataset_8ac564ef-6090-4f7c-9bdc-51865b2a86e2_files:/tmp/tmpn5toshnh/files/8/a/c/dataset_8ac564ef-6090-4f7c-9bdc-51865b2a86e2.dat']
```

since `upload.py` is used I guessed that it is `upload.xml`.

But, if I get you correctly then there are different ways to get data into Galaxy. If so we should also add analogous tests/fixes.

Reply:
> I'm surprised Planemo even uses upload.xml and not data fetch at this point.

@jmchilton I think upload happens here

https://github.com/galaxyproject/galaxy/blob/f9d0dabd2e97b74310f91c2e99892440520e930c/lib/galaxy/tool_util/verify/interactor.py#L485

Can you tell me more about this data fetch mechanism?